### PR TITLE
small improvment on network UX

### PR DESCRIPTION
**Pull request by vieux**

### CODE REVIEW

**Comment by vieux** [comment]
This fix the jenkins issue . I'd like to get it merged asap. ping @docker/swarm-maintainers

**Comment by jimmyxian** [comment]
LGTM :+1:

**Comment by chanwit** [comment]
LGTM

File: cluster/swarm/cluster.go [hunk 1]
```diff
@@ -176,7 +176,9 @@ func (c *Cluster) RemoveContainer(container *cluster.Container, force, volumes b
 
 // RemoveNetwork removes a network from the cluster
 func (c *Cluster) RemoveNetwork(network *cluster.Network) error {
-	return network.Engine.RemoveNetwork(network)
+	err := network.Engine.RemoveNetwork(network)
+	c.refreshNetworks()
```
Comment (aluzzardi) [inline cluster/swarm/cluster.go]: Uhm, why are we refreshing the entire cluster?

Comment (aluzzardi) [inline cluster/swarm/cluster.go]: Also, don't we usually do that in the proxy code rather than here? (for instance with images)

Comment (aluzzardi) [inline cluster/swarm/cluster.go]: As in, `proxyContainerAndForceRefresh` for instance

**Comment by vieux** [comment]
unfortunately the replication is managed by libnetwork doing a network rm on 1 node can potentially remove a network on other nodes in the cluster. 

With images doing an action on one node cannot change a completely different node. 

Since there is no event emited or anything I decided to do a full refresh. 

Feel free to open a pr if you have another idea

**Comment by aluzzardi** [comment]
Is there a way to know the scope of a network ?

**Comment by vieux** [comment]
@aluzzardi we might want to range over the engines, remove the network from memory and let the "random" refresh pick up the change. Wdyt?

**Comment by vieux** [comment]
Since rc2 yes

**Pull request merged**